maven-builder-support/src/main/java/org/apache/maven/building/UrlSource.java

```diff
@@ -87,6 +87,6 @@ public boolean equals(Object obj) {
         }
 
         UrlSource other = (UrlSource) obj;
-        return this.url.equals(other.url);
+        return Objects.equals(url.toExternalForm(), other.url.toExternalForm());
```

Comment:
Why actually?

Reply:
url equality check performs hostname resolution

Reply:
Really? WTF?! Why not `#toAsciiString()`?

Reply:
Whatever, am just creating string out of both in same way and comparing the two.